src/main/java/io/tiledb/java/api/Query.java

```diff
@@ -161,6 +161,92 @@ public Query setSubarray(NativeArray subarray) throws TileDBError {
     return this;
   }
 
+  /**
+   * Adds a 1D range along a subarray dimension, which is in the form (start, end). The datatype of
+   * the range components must be the same as the type of the domain of the array in the query.
+   *
+   * @param dimIdx The index of the dimension to add the range to
+   * @param start The range start
+   * @param end The range end
+   * @return This query
+   * @throws TileDBError A TileDB exception
+   */
+  public Query addRange(int dimIdx, Object start, Object end) throws TileDBError {
+    Datatype dimType = array.getSchema().getDomain().getType();
```

Comment:
you need to wrap these in a try block as well

try (Schema schema = array.getSchema(); Domain = schema.getDomain()) {}`